compiler/src/dotty/tools/dotc/transform/init/Potentials.scala

```diff
@@ -163,26 +155,29 @@ object Potentials {
 
   extension (pot: Potential) def toPots: Potentials = Potentials.empty + pot
 
-  extension (ps: Potentials) def select (symbol: Symbol, source: Tree)(using Context): Summary =
-    ps.foldLeft(Summary.empty) { case ((pots, effs), pot) =>
+  extension (ps: Potentials) def select (symbol: Symbol, source: Tree, selectEffect: Boolean = true)(using Context): Summary =
```

Comment:
I don't really get `selectEffect`'s purpose... :(

Reply:
Indeed we need a better name for that and add some documentation. It's used in potential expansion (`selectEffect 
 = false`), where we don't want to trigger a duplicate method call or field access effect.